cmd/kubelet/app/server.go

```diff
@@ -400,6 +407,33 @@ func UnsecuredDependencies(s *options.KubeletServer, featureGate featuregate.Fea
 	if err != nil {
 		return nil, err
 	}
+	var tp oteltrace.TracerProvider
+	if utilfeature.DefaultFeatureGate.Enabled(features.KubeletTracing) {
+	    hostname, err := nodeutil.GetHostname(s.HostnameOverride)
+	    if err != nil {
+		    return nil, err
+	    }
+	    opts := []otlpgrpc.Option{}
+		tc := componentbaseconfigv1alpha1.TracingConfiguration{
+			SamplingRatePerMillion:  s.KubeletConfiguration.Tracing.SamplingRatePerMillion,
+		}
+	    resourceOpts := []otelsdkresource.Option{
+		otelsdkresource.WithAttributes(
+			semconv.ServiceNameKey.String(hostname),
+			semconv.ServiceInstanceIDKey.String(componentKubelet),
+		),
+	    }
+		if s.KubeletConfiguration.Tracing.Endpoint != nil {
+			tc.Endpoint = s.KubeletConfiguration.Tracing.Endpoint
+	        opts = append(opts, otlpgrpc.WithEndpoint(*tc.Endpoint))
+		}
+	    opts = append(opts, otlpgrpc.WithInsecure())
+		tp, err = traces.InitTracing(context.Background(), tc, opts, resourceOpts)
+		if err != nil {
+			return nil, err
+		}
+		otel.SetTextMapPropagator(traces.Propagators())
```

Comment:
Setting the global textmappropagator shouldn't be necessary.  On the http and grpc handlers, we just need to pass a WithPropagators (or something) option explicitly.